crates/core/src/client/config.rs

```diff
@@ -22,6 +22,25 @@ pub enum Network {
     ProviderUrl(Url),
 }
 
+impl Network {
+    pub fn get_gateway_url(&self) -> Result<Url, ConfigError> {
+        match self {
+            Network::Mainnet => Ok(Url::parse("https://alpha-mainnet.starknet.io/feeder_gateway/").unwrap()),
+            Network::Goerli1 => Ok(Url::parse("https://alpha4.starknet.io/feeder_gateway/").unwrap()),
+            _ => Err(ConfigError::InvalidNetwork(format!("Network {:?} is not supported for provider url", self))),
```

Comment:
It doesn't work with goerli2?

Reply:
I couldn't find the gateway url for it online, will ask Lucas

Reply:
we had before, Starknet-rs already hard code these somewhere

```
        Network::Goerli1 => {
            StarknetProvider::SequencerGatewayProvider(SequencerGatewayProvider::starknet_alpha_goerli())
        }

        Network::Goerli2 => {
            StarknetProvider::SequencerGatewayProvider(SequencerGatewayProvider::starknet_alpha_goerli_2())
        }

        Network::Mainnet => {
            StarknetProvider::SequencerGatewayProvider(SequencerGatewayProvider::starknet_alpha_mainnet())
        }
```

Reply:
found it